space/virtual.go

```diff
@@ -0,0 +1,118 @@
+package space
+
+import (
+	"context"
+	"time"
+
+	"github.com/anyproto/any-sync/commonspace"
+	"github.com/anyproto/any-sync/commonspace/headsync"
+	"github.com/anyproto/any-sync/commonspace/object/acl/syncacl"
+	"github.com/anyproto/any-sync/commonspace/object/treesyncer"
+	"github.com/anyproto/any-sync/commonspace/objectsync"
+	"github.com/anyproto/any-sync/commonspace/objecttreebuilder"
+	"github.com/anyproto/any-sync/commonspace/spacestorage"
+	"github.com/anyproto/any-sync/commonspace/spacesyncproto"
+	"github.com/anyproto/any-sync/commonspace/syncstatus"
+	"github.com/anyproto/any-sync/net/peer"
+)
+
+type VirtualSpace struct {
+	*space
+}
+
+func NewVirtualSpace(s *service, spaceID string) *VirtualSpace {
+	coreSpace := newCommonSpace(spaceID)
+	vs := &VirtualSpace{
+		space: &space{
+			service:                s,
+			Space:                  coreSpace,
+			installer:              s.bundledObjectsInstaller,
+			loadMandatoryObjectsCh: make(chan struct{}),
+		},
+	}
+	return vs
+}
+
+func (vs *VirtualSpace) Close(_ context.Context) error {
+	staticObjects := vs.service.sourceService.GetStaticObjectsBySpaceID(vs.Id())
+	for _, id := range staticObjects {
+		vs.service.sourceService.RemoveStaticSource(id)
```

Comment:
it will work only on graceful shutdown

Reply:
May be we should register virtual spaces in object store and query them on Anytype start? So if virtual space exists in object store, we should check its objects and remove them